### PR TITLE
Rename doc to docs and fix minor language issues

### PR DESCRIPTION
**Pull request by spydon**
# Description

* Renames doc to docs
* Fixes and updates some grammar and words in the docs

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `develop` (not `master`)
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: docs/input.md [hunk 1]
```diff
@@ -123,7 +123,7 @@ class TapableComponent extends PositionComponent with Tapable {
 
 Flame provides a simple way to access Flutter's features regarding accessing Keyboard input events.
 
-To use it just add the `KeyboardEvents` mixin to your game class. By doing so you will need to implement the `onKeyEvent` method, this method is called every time a keyboard event happens, and it receives an instance of the Flutter class `RawKeyEvent`, which can be used to get information about the event, like if it was a key down or key up event, the pressed key and etc.
+To use it just add the `KeyboardEvents` mixin to your game class. By doing so you will need to implement the `onKeyEvent` method, this method is called every time a keyboard event happens, and it receives an instance of the Flutter class `RawKeyEvent`. The event can be used to get information about what occurred, like if it was a key down or key up event, a pressed key etc.
```
Comment (awulkan) [inline docs/input.md:126]: I believe that the original text specified that you got information about which key was pressed, not just that a key was pressed.

File: docs/widgets.md [hunk 2]
```diff
@@ -14,28 +14,28 @@ The grid sprite is a 3x3 grid and with 9 blocks, representing the 4 corners, the
...
+        tileSize: 16,        // The width/height of the tile on your grid image
+        destTileSize: 50,    // The dimensions to be used when drawing the tile on the canvas
         child: SomeWidget(), // Any Flutter widget
 )
 ```
 
 ## SpriteButton
 
-`SpriteButton` is a simple widget that create a button based on Flame sprites, this can be very usefull when trying to create retro looking buttons for example, where creating that look can be easy to achieve drawing the button on a drawing software instead of making it directly on Flutter.
+`SpriteButton` is a simple widget that creates a button based on Flame sprites. This can be very useful when trying to create retro looking buttons. For example when creating that look can be easy to achieve when drawing the button in a drawing software, instead of making it directly in Flutter.
```
Comment (awulkan) [inline docs/widgets.md:34]: > For example when creating that look can be easy to achieve when drawing the button in a drawing software, instead of making it directly in Flutter.

I would probably have said:
> For example when creating that look can be more easily achievable by drawing the button in a drawing software, instead of making it directly in Flutter.

Just to make the sentence a bit easier to understand.

**Comment by awulkan** [comment]
I've taken a quick look at this PR and left some comments. Overall it looks very good and I think it will help make the documentation a lot clearer.

I'm on my phone so I didn't review everything.

File: docs/input.md (hunk 1 above)
Comment (spydon) [reply docs/input.md:126]: The sentence is so long though, and has 5 commas. But I can try to restructure it so that it is closer to the original.

File: docs/audio.md [hunk 3]
```diff
@@ -1,6 +1,19 @@
 # Audio
 
-To play an audio, it's really simple! Just run, at any moment:
+To play an audio is really simple!
+
+Start by adding an entry in your `pubspec.yaml` for each file you want to use (and make sure at the files exist in path that you provide).
```
[the remark this answers is not in the document]
Comment (spydon) [reply docs/audio.md:5]: Indeed it should be, good catch!

File: docs/input.md (hunk 1 above)
Comment (awulkan) [reply docs/input.md:126]: The only thing that needs to be changed is this part "a pressed key etc.". To something like "which key was pressed, etc.".

File: docs/components.md [hunk 4]
```diff
@@ -43,14 +43,14 @@ The `onDestroy` method can be overridden to run code before the component is rem
...
-* The `AnimationComponent` takes an `Animation` object and renders a cyclic animated sprite (more details about Animations [here](/doc/images.md#Animation))
+* The `AnimationComponent` takes an `Animation` object and renders a cyclic animated sprite (more details about Animations [here](./images.md#Animation))
 * The `SvgComponent` takes an `Svg` object and renders the SVG on the game
 * The `ParallaxComponent` can render a parallax background with several frames
 * The `Box2DComponent`, that has a physics engine built-in (using the [Box2D](https://github.com/google/box2d.dart) port for Dart)
 
 ## Animation Component
 
-This component uses an instance of the [Animation](/doc/images.md#Animation) class to represent a Component that has a sprite that runs a single cyclic animation.
+This component uses an instance of the [Animation](./images.md#Animation) class to represent a Component that has a sprite that runs a single cyclic animation.
```
Comment (erickzanardo) [inline docs/components.md:53]: we probably need to keep the old format, as there is a script that reads this files and generate the site based on it and I don't know if this will work.

@luanpotter what do you think? since you made the script?

Comment (spydon) [reply docs/components.md:53]: Where does this script reside? :)

Comment (erickzanardo) [reply docs/components.md:53]: It is on the flame engine site repository, under the folder generate docs, I would get you a link, but the GH mobile app still has lots to improve hahhahaha

Comment (spydon) [reply docs/components.md:53]: Haha I can probably find it from that description.

Comment (spydon) [reply docs/components.md]: Reverted to use the full paths

Comment (luanpotter) [reply docs/components.md]: https://github.com/flame-engine/flame-engine-site/blob/master/cmds/generate_docs.sh

yes we do need the full path. also changing the folder name might require a second PR on the other repo

Comment (spydon) [reply docs/components.md]: Did a PR for that repository too:
https://github.com/flame-engine/flame-engine-site/pull/1

File: docs/palette.md [hunk 5]
```diff
@@ -2,11 +2,11 @@
...
+The `Color` class is nothing but a wrapper over a single simple ARGB color in hexadecimal integer format. So to create a Color object you just need to pass in the color as an integer in the ARGB format.
 
-You can use Dart's hexadecimal notation to make it really easy; for instance: `0xFF00FF00` is fully opaque green (the 'mask' would be `0xAARRGGBB`). Do note the first two hexadecimal digits are for the alpha channel (transparency), unlike on regular (non-A) RGB. The max(FF = 256) means fully opaque.
+You can use Dart's hexadecimal notation to make it really easy; for instance: `0xFF00FF00` is fully opaque green (the 'mask' would be `0xAARRGGBB`). Do note the first two hexadecimal digits are for the alpha channel (transparency), unlike on regular (non-A) RGB. The max(FF = 255) means fully opaque.
 
-There is a color enum to make it easy to use common colors; it is in the material flutter package:
+There is a color enum to make it easy to use common colors, it resides in the material flutter package:
```
Comment (luanpotter) [inline docs/palette.md:9]: I actually think this should still be a `;` :p

File: docs/text.md [hunk 6]
```diff
@@ -4,25 +4,30 @@ Flame has some dedicated classes to help you render text.
...
 # TextConfig
 
-A Text Config contains all typographical information required to render texts; i.e., font size and color, family, etc.
+A Text Config contains all typographical information required to render text; i.e., font size and color, family, etc.
 
 Example usage:
 
 ```dart
-const TextConfig config = TextConfig(fontSize: 48.0, fontFamily: 'Awesome Font', anchor: Anchor.rightBottom);
+const TextConfig config = TextConfig(fontSize: 48.0, fontFamily: 'Awesome Font');
```
Comment (luanpotter) [inline docs/text.md:12]: why did you remove the anchor example?

Comment (luanpotter) [reply docs/text.md:12]: oh gotcha

File: docs/widgets.md [hunk 7]
```diff
@@ -14,28 +14,28 @@ The grid sprite is a 3x3 grid and with 9 blocks, representing the 4 corners, the
...
 ```dart
 import 'package:flame/widgets/nine_tile_box.dart';
 
 NineTileBox(
-        image: image, // dart:ui image instance
-        tileSize: 16, // The width/height of the tile on your grid image
-        destTileSize: 50, // The dimensions to be used when drawing the tile on the canvas
+        image: image,        // dart:ui image instance
+        tileSize: 16,        // The width/height of the tile on your grid image
+        destTileSize: 50,    // The dimensions to be used when drawing the tile on the canvas
```
Comment (luanpotter) [inline docs/widgets.md:27]: I actually prefer the comments close to the code, I think it looks odd with spaces extra spacing

File: lib/text_config.dart [hunk 8]
```diff
@@ -70,8 +70,8 @@ class TextConfig {
   ///
   /// Example usage:
   ///
-  ///     const TextConfig config = TextConfig(fontSize: 48.0, fontFamily: 'Awesome Font', anchor: Anchor.rightBottom);
-  ///     config.render(c, Offset(size.width - 10, size.height - 10);
+  ///     const TextConfig config = TextConfig(fontSize: 48.0, fontFamily: 'Awesome Font');
+  ///     config.render(c, Offset(size.width - 10, size.height - 10, anchor: Anchor.bottomRight);
```
Comment (luanpotter) [inline lib/text_config.dart:74]: awesome catch!

File: docs/palette.md (hunk 5 above)
Comment (spydon) [reply docs/palette.md:9]: Hmm, I it works, but it's not really the proper use of the semi-colon in English. It is not perfect with the comma either though. Probably should have rewritten the sentence instead. 😅

File: lib/text_config.dart (hunk 8 above)
Comment (spydon) [reply lib/text_config.dart:74]: It was someone that reported it in the Discord channel. :)

File: docs/widgets.md (hunk 7 above)
Comment (spydon) [reply docs/widgets.md]: Fixed!

File: docs/palette.md (hunk 5 above)
Comment (spydon) [reply docs/palette.md]: Fixed!
